staging/src/k8s.io/kubectl/pkg/polymorphichelpers/history.go

```diff
@@ -48,6 +48,7 @@ const (
 // HistoryViewer provides an interface for resources have historical information.
 type HistoryViewer interface {
 	ViewHistory(namespace, name string, revision int64) (string, error)
+	GetHistory(namespace, name string) (map[int64]runtime.Object, error)
```

Comment:
I understand that we can not easily remove `ViewHistory`. But just curiosity; semantically, GetHistory also **contains** `ViewHistory`?.

Reply:
I mean consumers can do the same behavior just using the `GetHistory` without needing `ViewHistory`

Reply:
GetHistory is just to get the data associated with the history, while ViewHistory gets *and* prints the history.

You are correct, a consumer can use GetHistory alone if they want to control how the data is handled.  Or they can use ViewHistory alone if they want to let it get and print the output in the predefined way.

---

I thought about trying to refactor ViewHistory, but decided it would be too big of a change for this PR.  I do think it probably needs to be refactored though.  Maybe as a follow-up task.

Ideally, I think it would be good for ViewHistory to only be concerned with printing the history, and that it should not need to know how to also retrieve the history data.  Maybe even move it to a describer like some of the existing TODO comments suggest.